packages/bs-neo-legacy/src/__tests__/BSNeoLegacy.spec.ts

```diff
@@ -217,4 +217,83 @@ describe('BSNeoLegacy', () => {
+      serviceNeoLegacy = new BSNeoLegacy('neoLegacy', BSNeoLegacyConstants.DEFAULT_NETWORK)
+
+      const account = serviceNeoLegacy.generateAccountFromKey(keyEmptyBalance)
+
+      calculateParams = { account }
+      migrateParams = { account, address: 'NXLMomSgyNeZRkeoxyPVJWjSfPb7xeiUJD' }
+    })
+
+    it("Shouldn't be able to migrate to Neo 3 when it's using a non Mainnet network", async () => {
+      serviceNeoLegacy = new BSNeoLegacy('neoLegacy', network)
+
+      await expect(serviceNeoLegacy.migrateToNeo3(migrateParams)).rejects.toThrow('Must use Mainnet network')
+    })
+
+    it("Shouldn't be able to migrate to Neo3 when it's not using an address", async () => {
+      migrateParams.address = ''
+
+      await expect(serviceNeoLegacy.migrateToNeo3(migrateParams)).rejects.toThrow('Must have address')
+    })
+
+    it("Shouldn't be able to migrate to Neo 3 when it's not using GAS or NEO enough", async () => {
+      await expect(serviceNeoLegacy.migrateToNeo3(migrateParams)).rejects.toThrow('Must have at least 0.1 GAS or 2 NEO')
+    })
+
+    it("Shouldn't be able to calculate values to migrate to Neo 3 when it's using a non Mainnet network", async () => {
+      serviceNeoLegacy = new BSNeoLegacy('neoLegacy', network)
+
+      await expect(serviceNeoLegacy.calculateToMigrateToNeo3Values(migrateParams)).rejects.toThrow(
+        'Must use Mainnet network'
+      )
+    })
+
+    it("Shouldn't be able to calculate values to migrate to Neo 3 when it's not using GAS or NEO enough", async () => {
+      await expect(serviceNeoLegacy.calculateToMigrateToNeo3Values(calculateParams)).rejects.toThrow(
+        'Must have at least 0.1 GAS or 2 NEO'
+      )
+    })
+
+    it('Should be able to calculate values to migrate to Neo 3 when pass the correct params', async () => {
+      const account = serviceNeoLegacy.generateAccountFromKey(keyWithBalance)
+
+      calculateParams = { account }
+
+      const response = await serviceNeoLegacy.calculateToMigrateToNeo3Values(calculateParams)
+
+      expect(response).toEqual({
+        gasFee: expect.any(String),
+        gasAmount: expect.any(String),
+      })
+    })
+
+    it('Should throw an error when migrate to Neo 3 is from an empty balance address', async () => {
+      const account = serviceNeoLegacy.generateAccountFromKey(keyEmptyBalance)
+
+      migrateParams = { account, address: 'NQTRnAv45ax2avMDBjmEKEyNmc4Ms9zLjH' }
+
+      await expect(serviceNeoLegacy.migrateToNeo3(migrateParams)).rejects.toThrow()
+    })
+
+    it.skip('Should create a successfully migration when migrate to Neo 3 is called with correct GAS params', async () => {
+      const account = serviceNeoLegacy.generateAccountFromKey(keyWithBalance)
+
+      migrateParams = { account, address: 'NWs1nPJP8XkZxsNwuGYmikzQrEJcZBAUJN' }
+
+      const [txId] = await serviceNeoLegacy.migrateToNeo3(migrateParams)
+
+      expect(typeof txId).toBe('string')
+      expect(txId.length).toBeGreaterThan(0)
+    })
+  })
```

Comment:
What do you think of creating a new file or just move it outside of the container describe ?

Just to make it more organized